src/hpke.rs

```diff
@@ -0,0 +1,602 @@
+// Copyright (C) Nitrokey GmbH
+// SPDX-License-Identifier: Apache-2.0 or MIT
+
+use crate::StagingBackend;
+
+use trussed::{
+    config::MAX_SERIALIZED_KEY_LENGTH,
+    key,
+    serde_extensions::ExtensionImpl,
+    service::Filestore,
+    store::keystore::Keystore,
+    types::{KeyId, Message},
+    Bytes,
+};
+use trussed_hpke::*;
+
+type HkdfSha256 = hkdf::Hkdf<sha2::Sha256>;
+type HkdfSha256Extract = hkdf::HkdfExtract<sha2::Sha256>;
+
+use rand_core::{CryptoRng, RngCore};
+use salty::agreement as x25519;
+
+const X25519_KEM_SUITE_ID: &[u8] = b"KEM\x00\x20";
+const X25519_HKDF_CHACHA20_POLY1305_HPKE_SUITE_ID: &[u8] = b"HPKE\x00\x20\x00\x01\x00\x03";
```

Comment:
nitpicky: as this is very verbose already, shouldn't this then also contain `SHA256` - the var name I mean ?

Reply:
Right.

I fixed it.